membership-attribute-service/app/services/SupporterRatePlanToAttributesMapper.scala

```diff
@@ -0,0 +1,321 @@
+        "2c92c0f86fa49142016fa49ea0d028b6",
+        "2c92c0f86fa49142016fa49e9b9a286f",
+        "2c92c0f86fa49142016fa49ea56a2938",
+        "2c92c0f861f9c26d0161fc434bfe004c",
+        "2c92c0f95aff3b56015b1045fb9332d2",
+        "2c92c0f8555ce5cf01556e7f01b81b94",
+        "2c92c0f8555ce5cf01556e7f01771b8a",
+        "2c92c0f9555cf10501556e84a70440e2",
+        "2c92c0f961f9cf300161fc4d2e3e3664",
+        "2c92c0f85aff3453015b1041dfd2317f",
+        "2c92c0f955c3cf0f0155c5d9df433bf7",
+        "2c92c0f955c3cf0f0155c5d9ddf13bc5",
+        "2c92c0f955c3cf0f0155c5d9e2493c43"
+      ) -> paperTransformer,
+      List(
+        "2c92c0f86fa49142016fa49eb1732a39",
+        "2c92c0f86fa49142016fa49ea90e2976",
+        "2c92c0f86fa49142016fa49eaecb29dd",
+        "2c92c0f86fa49142016fa49ea1af28c8",
+        "2c92c0f86fa49142016fa49eaa492988",
+        "2c92c0f961f9cf300161fc44f2661258",
+        "2c92c0f955a0b5bf0155b62623846fc8",
+        "2c92c0f95aff3b54015b1047efaa2ac3",
+        "2c92c0f855c3b8190155c585a95e6f5a",
+        "2c92c0f95aff3b53015b10469bbf5f5f",
+        "2c92c0f961f9cf300161fc4f71473a34",
+        "2c92c0f955c3cf0f0155c5d9e83a3cb7",
+        "2c92c0f95aff3b56015b104aa9a13ea5",
+        "2c92c0f85aff33ff015b1042d4ba0a05",
+        "2c92c0f85aff3453015b10496b5e3d17"
+      ) -> paperPlusDigitalTransformer,
+      List(
+        "2c92c0f965f2122101660fb33ed24a45",
+        "2c92c0f967caee410167eff78e7b5244",
+        "2c92c0f965f2122101660fb81b745a06",
+        "2c92c0f96df75b5a016df81ba1c62609",
+        "2c92c0f965d280590165f16b1b9946c2",
+        "2c92c0f867cae0700167eff921734f7b",
+        "2c92c0f965dc30640165f150c0956859",
+        "2c92c0f96ded216a016df491134d4091"
+      ) -> guardianWeeklyTransformer,
+      List(
+        "2c92c0f94c9ca1c5014c9e5c64ba4260",
+        "2c92c0f945fee1c90146057402c7066b"
+      ) -> memberTransformer("Friend"),
+      List(
+        "2c92c0f849c6e58a0149c73d6f114be2"
+      ) -> memberTransformer("Staff"),
+      List(
+        "2c92c0f94c510a0d014c569ba8eb45f7",
+        "2c92c0f94c510a01014c569e2d857cfd",
+        "2c92c0f84b079582014b2754c07c0f7d",
+        "2c92c0f84b079582014b2754bfd70f6d"
+      ) -> memberTransformer("Supporter"),
+      List(
+        "2c92c0f94c510a0d014c569a93194575",
+        "2c92c0f84c510081014c569a18b04e84",
+        "2c92c0f945fee1c9014605749e450969",
+        "2c92c0f8471e22bb01471ffe9596366c"
+      ) -> memberTransformer("Partner"),
+      List(
+        "2c92c0f84c5100b6014c56908a63216d",
+        "2c92c0f94c510a04014c568d648d097d",
+        "2c92c0f845fed48301460578277167c3",
+        "2c92c0f9471e145d01471ffd7c304df9"
+      ) -> memberTransformer("Patron")
+    )
+  )
+
+  def digitalSubTransformer(attributes: Attributes, supporterRatePlanItem: SupporterRatePlanItem) =
+    attributes.copy(
+      DigitalSubscriptionExpiryDate = Some(
+        getLatestDate(attributes.DigitalSubscriptionExpiryDate, supporterRatePlanItem.termEndDate)
+      )
+    )
```

Comment:
the SAM thing would be add a trait like this 
```scala
  trait SubscriptionTransformer {
    def transform(attributes: Attributes, supporterRatePlanItem: SupporterRatePlanItem): Attributes
  }
```
then make it a val and add the trait type to the val
```suggestion
  val digitalSubTransformer: SubscriptionTransformer = (attributes: Attributes, supporterRatePlanItem: SupporterRatePlanItem) =>
    attributes.copy(
      DigitalSubscriptionExpiryDate = Some(
        getLatestDate(attributes.DigitalSubscriptionExpiryDate, supporterRatePlanItem.termEndDate)
      )
    )

```